workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/InjectionConditionAttribute.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    /// <summary>
+    ///     Used to denote when a specific IInjectable type is ready to be created.
+    /// </summary>
+    internal enum InjectionCondition
```

Comment:
Shower thought question here: Wouldn't the injection conditions solely depend on the `InjectableType` of an injectable? I.e. A writer or command request handler will always require authority over a component. If that's the case, isn't this attribute here be a little redundant since you can infer the requirements from the InjectableType?

Reply:
Right now no - we have a bit of a confusing mess as a result of Writers being Readers to the extent that they are just one class. Readers and Writers to the same component have the exact same `InjectableId` as a result but with different conditions... We could make them their own thing by separating them in `InjectableType`, and creating separate `InjectableCreators` (which could even just do the same thing for now), but then the dispatchers would have to query separately for readers and writers whenever they pass things for user callbacks. Maybe that's a change worth making all in all?